### PR TITLE
Feat: add no-setter-return

### PR DESCRIPTION
**Pull request by magurotuna**
This PR adds `no-setter-return` rule, requested in #4 

Please merge this PR  _after_ #6, because some test cases cannot be passed without #6.

### CODE REVIEW

File: xtask/src/docgen/extract.rs [hunk 1]
```diff
@@ -204,6 +204,9 @@ impl Parse for RuleTests {
 
         ok_examples.truncate(30);
 
+        // If there is a trailing comma, consumes it.
+        let _ = input.parse::<Token!(,)>();
+
```
Comment (magurotuna) [inline xtask/src/docgen/extract.rs:209]: This change is related to #5, required to run `cargo xtask docgen` on `rule_test!` containing trailing comma.

File: rslint_core/src/groups/errors/no_setter_return.rs [hunk 2]
```diff
@@ -0,0 +1,363 @@
...
+        key: &SyntaxNode,
+        body: &SyntaxNode,
+        mut stmts: impl Iterator<Item = ast::Stmt>,
+        ctx: &mut RuleCtx,
+    ) {
+        if stmts.any(|stmt| self.stmt_returns_value(&stmt)) {
+            let err = ctx.err(
+                self.name(),
+                format!(
+                    "Setter properties are not allowed to return values, but `{}` does.",
```
Comment (RDambrosio016) [inline rslint_core/src/groups/errors/no_setter_return.rs:232]:
```suggestion
                    "setter properties are not allowed to return values, but `{}` does.",
```
Error messages should be lowercase

Comment (RDambrosio016) [reply rslint_core/src/groups/errors/no_setter_return.rs:232]: The error also needs to include a primary label to the error to help the user find out where it is and to be consistent

File: xtask/src/docgen/extract.rs [hunk 3]
```diff
@@ -204,6 +204,9 @@ impl Parse for RuleTests {
 
         ok_examples.truncate(30);
 
+        // If there is a trailing comma, consumes it.
```
Comment (RDambrosio016) [inline xtask/src/docgen/extract.rs:207]:
```suggestion
        // if there is a trailing comma, consume it.
```
Small nit

**Review by RDambrosio016 — CHANGES_REQUESTED** [review]
Thank you for the PR!
Looks good to me except for a few minor nitpicks

**Comment by magurotuna** [comment]
@RDambrosio016 thanks for the detailed review! Addressed all the points. Please check again :)

**Comment by RDambrosio016** [comment]
Looks good to me!

**Pull request merged**